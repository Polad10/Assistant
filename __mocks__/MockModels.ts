const appointment_1 = { id: 1, datetime: '2024-01-01 19:44:00.000Z', actions: null, treatment_id: 1 }
const appointment_2 = { id: 2, datetime: '2024-01-02 19:44:00.000Z', actions: null, treatment_id: 2 }

const patient_1 = {
  id: 1,
  first_name: 'FirstName',
  last_name: 'LastName',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const patient_2 = {
  id: 2,
  first_name: 'FirstName2',
  last_name: 'LastName2',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const payment_1 = {
  id: 1,
  date: '2024-01-01 19:44:00.000Z',
  amount: 100,
  treatment_id: 1,
}

const payment_2 = {
  id: 2,
  date: '2024-01-02 19:44:00.000Z',
  amount: 100,
  treatment_id: 2,
}

const treatment_1 = {
  id: 1,
  start_date: '2024-01-01 19:44:00.000Z',
  end_date: null,
  title: 'Title',
  patient_id: 1,
  price: 100,
}

const treatment_2 = {
  id: 2,
  start_date: '2024-01-02 19:44:00.000Z',
  end_date: null,
  title: 'Title2',
  patient_id: 2,
  price: 100,
}

const setting = {
  language: 'en',
}

export const mockModels = {
  appointment_1,
  appointment_2,
  patient_1,
  patient_2,
  payment_1,
  payment_2,
  treatment_1,
  treatment_2,
  setting,
}
