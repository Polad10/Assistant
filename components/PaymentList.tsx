import { Colors } from '../types/Colors'
import { View, StyleSheet, ScrollView } from 'react-native'
import PaymentItem from './PaymentItem'
import { useTheme } from '@react-navigation/native'
import { Divider } from '@rneui/themed'
import { RootStackParamList } from '../types/Navigation'
import { Payment } from '../modals/Payment'
import MainView from './MainView'

type Props = {
  pageName: keyof RootStackParamList
  payments: Payment[]
}

export default function PaymentList(props: Props) {
  const { colors } = useTheme()

  const paymentElements = props.payments.map((p) => {
    return (
      <View key={p.id}>
        <PaymentItem payment={p} pageName={props.pageName} />
        <Divider color={colors.border} style={styles(colors).divider} />
      </View>
    )
  })

  function getPaymentsContentView() {
    return (
      <ScrollView keyboardDismissMode='on-drag'>
        <MainView>{paymentElements}</MainView>
      </ScrollView>
    )
  }

  return <MainView>{getPaymentsContentView()}</MainView>
}

const styles = (colors: Colors) =>
  StyleSheet.create({
    divider: {
      marginHorizontal: 13,
    },
  })
