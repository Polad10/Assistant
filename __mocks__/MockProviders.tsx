import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
