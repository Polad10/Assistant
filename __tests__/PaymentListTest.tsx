import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import PaymentList from '../components/PaymentList'
import { mockModels } from '../__mocks__/MockModels'

describe('PaymentList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <PaymentList pageName='Patient' payments={[mockModels.payment_1, mockModels.payment_2]} />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
