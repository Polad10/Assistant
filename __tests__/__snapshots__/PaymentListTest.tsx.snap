// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentList renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    data={
      [
        {
          "amount": 100,
          "date": "2024-01-01 19:44:00.000Z",
          "id": 1,
          "treatment_id": 1,
        },
        {
          "amount": 100,
          "date": "2024-01-02 19:44:00.000Z",
          "id": 2,
          "treatment_id": 2,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    keyboardDismissMode="on-drag"
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
    waitFor={
      [
        {
          "current": null,
        },
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#121212",
                    "borderColor": "#bcbbc1",
                    "flexDirection": "row",
                    "padding": 14,
                  }
                }
                testID="RNE__LISTITEM__padView"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                        "justifyContent": "center",
                      },
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        {
                          "justifyContent": "space-between",
                        },
                      ],
                    ]
                  }
                  theme={
                    {
                      "colors": {
                        "background": "#ffffff",
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": {
                          "android": {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ad1457",
                        "success": "#52c41a",
                        "warning": "#faad14",
                        "white": "#ffffff",
                      },
                      "spacing": {
                        "lg": 12,
                        "md": 8,
                        "sm": 4,
                        "xl": 24,
                        "xs": 2,
                      },
                    }
                  }
                >
                  <View>
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "#faf8ff",
                          "fontSize": 17,
                        }
                      }
                      testID="listItemTitle"
                    >
                      Title
                    </Text>
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "#faf8ff",
                          "fontSize": 15,
                          "marginTop": 10,
                          "opacity": 0.5,
                        }
                      }
                      testID="listItemTitle"
                    >
                      Invalid DateTime
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "lightgreen",
                          "fontSize": 17,
                          "marginRight": 10,
                        }
                      }
                      testID="listItemTitle"
                    >
                      + 
                      100
                    </Text>
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "borderBottomColor": "#374151",
                "borderBottomWidth": 0.5,
                "marginHorizontal": 10,
              }
            }
            testID="RNE__Divider"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          />
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#121212",
                    "borderColor": "#bcbbc1",
                    "flexDirection": "row",
                    "padding": 14,
                  }
                }
                testID="RNE__LISTITEM__padView"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                        "justifyContent": "center",
                      },
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        {
                          "justifyContent": "space-between",
                        },
                      ],
                    ]
                  }
                  theme={
                    {
                      "colors": {
                        "background": "#ffffff",
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": {
                          "android": {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ad1457",
                        "success": "#52c41a",
                        "warning": "#faad14",
                        "white": "#ffffff",
                      },
                      "spacing": {
                        "lg": 12,
                        "md": 8,
                        "sm": 4,
                        "xl": 24,
                        "xs": 2,
                      },
                    }
                  }
                >
                  <View>
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "#faf8ff",
                          "fontSize": 17,
                        }
                      }
                      testID="listItemTitle"
                    >
                      Title2
                    </Text>
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "#faf8ff",
                          "fontSize": 15,
                          "marginTop": 10,
                          "opacity": 0.5,
                        }
                      }
                      testID="listItemTitle"
                    >
                      Invalid DateTime
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "lightgreen",
                          "fontSize": 17,
                          "marginRight": 10,
                        }
                      }
                      testID="listItemTitle"
                    >
                      + 
                      100
                    </Text>
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "borderBottomColor": "#374151",
                "borderBottomWidth": 0.5,
                "marginHorizontal": 10,
              }
            }
            testID="RNE__Divider"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
